Fix sync docs toc (#3158)
kyma-project/kyma-environment-broker: docs/contributor/04-10-workflows.md

The [`sync-docs-toc`](/.github/workflows/sync-docs-toc.yml) workflow automatically opens a PR to the `product-kyma-runtime` repository whenever a new or renamed file in KEB's `docs/` directory contains the `<!--{"metadata":{"publish":true}}-->` metadata.
These documents are published to the Restricted Markets documentation. Content edits to existing files require no action — `product-kyma-runtime` pulls the latest content automatically using a `fileTree` reference in its `manifest.yaml`.

> The workflow is triggered after every successful [Create and Promote Release](#create-and-promote-release-workflow) or [Promote KEB to DEV](#promote-keb-to-dev-workflow) workflow run. It can also be triggered manually via `workflow_dispatch`.

Reviewer: ```suggestion
The workflow is triggered after every successful [Create and Promote Release](#create-and-promote-release-workflow) or [Promote KEB to DEV](#promote-keb-to-dev-workflow) workflow run. You can also trigger it manually using `workflow_dispatch`.
```

The workflow performs the following steps:

1. Resolves the diff range using the two most recent GitHub releases (`gh release list`)
2. Detects added or renamed `.md` files in `docs/` between the two releases, filtered by `publish:true` metadata
3. Fetches the corresponding chart version from `management-plane-charts` (falls back to searching merged PRs if the release branch was already deleted)
4. Updates `docs/toc.yaml` in `product-kyma-runtime` using the [`scripts/python/update_toc.py`](../../scripts/python/update_toc.py) script, which inserts new entries in numeric-prefix order and updates the renamed ones
5. Opens a PR to `product-kyma-runtime` with the title `docs(keb): sync doc structure changes [chart@<version>]`

If no relevant changes are detected, the workflow exits early and no PR is created. On failure, a Slack notification is sent to `kyma-gopher-private-alerts`.

> ### Manual trigger

Reviewer: ```suggestion
### Manual Trigger
```

> The workflow can be triggered manually with the following inputs:

Reviewer: ```suggestion
You can manually trigger the workflow with the following inputs:
```

| Input | Required | Description |
|-------|:--------:|-------------|
| **`prev_tag`** | yes | Older release tag to use as the base of the diff (e.g. `1.29.13`) |

> | **`current_tag`** | no | Newer release tag to diff against. Defaults to the latest published release. |

Reviewer: ```suggestion
| `prev_tag` | yes | Older release tag to use as the base of the diff (for example, `1.29.13`) |
| `current_tag` | no | Newer release tag to diff against. Defaults to the latest published release. |
```
